cmake/libcudacxxInstallRules.cmake

```diff
@@ -0,0 +1,38 @@
+option(libcudacxx_ENABLE_INSTALL_RULES
+  "Enable installation of libcudacxx" ${libcudacxx_TOPLEVEL_PROJECT}
+)
+
+if (NOT libcudacxx_ENABLE_INSTALL_RULES)
+  return()
+endif()
+
+# Bring in CMAKE_INSTALL_LIBDIR
+include(GNUInstallDirs)
+
+# libcudacxx is a header library; no need to build anything before installing:
+set(CMAKE_SKIP_INSTALL_ALL_DEPENDENCY TRUE)
```

Comment:
This logic seems to be tied to the fact that `libcudacxx` isn't the top project and not the fact we are asking for install rules. We need to either guard this with `TOPLEVEL_PROJECT` or move it to where we set `TOPLEVEL_PROJECT`

Reply:
My understanding is that this flag just disables the dependency on building `all` when doing `make install`, and since we're just setting a local variable, this effect should be limited to the libcudacxx install rules (AIUI).

The install rules are disabled by default when `TOPLEVEL` is false, so this shouldn't impact any users unless they opt-in. I want to have this available to subprojects since it might be useful for generating temporary Thrust installs during testing.

Is there an issue I'm missing?

Reply:
Since this is inside an `include()` at the root of the project this variable is now defined for all subdirectories of the repository. Which means that when building this is the top/root project this value will be set for `add_subdirectory(libcxx)` and the subsequent creation of the `libcxx` and `libcxx_static` libraries.

Reply:
That's a good point, makes sense. I'll just take this out for now, since it's just a "nice to have." If we can find a way to set this per-project I'll add it back.